### DISCUSSION TITLE
How to connect keycloak with keycloakify ?

### DISCUSSION THREAD

**Discussion by Ramnirmal0** [keycloakify/keycloakify#60] [Q&A] [answered]
Hi @garronej , 

Thanks for contributing keycloakify .

I am new to keycloak and keycloakify so when i gone through the documentation of keycloakify i am confused about how to connect keycloak with keyclokify ? where do i have to mention the keycloak config details ( realm , clientid , resource_name , url... or keycloak.json file). 
where do i have to add [this code](https://github.com/InseeFrLab/keycloakify#implement-context-persistence-optional)?

could you please guide me.

**Comment by garronej** [comment, marked as answer]
Hi @Ramnirmal0,  

> where do i have to mention the keycloak config details ( realm , clientid , resource_name , url... or keycloak.json file)

You don't have to. Keycloakify generates a `.jar` file that you provide to Keycloak. This `.jar` file contains your therme. Once you have imported it in Keycloak you just need to select it in the administration panel.  

There is [a video](https://github.com/InseeFrLab/keycloakify#kickstart-video) and [a minimal setup](https://github.com/garronej/keycloakify-demo-app) you can use.  

Follow the steps and you will see, information about how to proceed printed in the console.

**Reply by Ramnirmal0** [reply]
thank you @garronej  . one more thing i like to know how to change css of login page. where should i make changes / specify the css ?

**Reply by Ramnirmal0** [reply]
@garronej 
i like to change body background color and provide some border radius to login form . [ assume dark theme ] . where should I make changes in demo boilerplate ?

**Reply by garronej** [reply]
You have to provides classnames as props to the `<KcApp>` component.  

[Theses](https://github.com/InseeFrLab/keycloakify/blob/99a5efe36c3a78b5c00f955dbe89101ce89da501/src/lib/components/KcProps.ts#L126-L196) are the classes that are applied by default.

Here for example we set the header in red by doing this:  

![image](https://user-images.githubusercontent.com/6702424/145537839-a4703b2a-d7c6-4ee6-9fe4-d911b9338396.png)

In the example I use [tss-react](https://github.com/garronej/tss-react) but you could as well have a `.css` file like:  

`index.css`
```css

.foo {
    color: red;
}

```

`index.tsx`
```tsx
import "./index.css";
import { App } from "./<wherever>/App";
import { KcApp, defaultKcProps, getKcContext } from "keycloakify";
import { css } from "tss-react/@emotion/css";

const { kcContext } = getKcContext();

const myClassName = css({ "color": "red" });

reactDom.render(
    <KcApp
        kcContext={kcContext}
        {...{
            ...defaultKcProps,
            "kcHeaderWrapperClass": "foo",
        }}
    />,
    document.getElementById("root"),
);
```

I advise you to use `tss-react` though. It's much more easy to maintain.  

[Here](https://github.com/InseeFrLab/onyxia-web/blob/fa4b7ec694c78803e1aaf55b7c8843d2b6baaebf/src/app/components/KcApp/KcApp.tsx#L40) and [here](https://github.com/InseeFrLab/onyxia-web/blob/fa4b7ec694c78803e1aaf55b7c8843d2b6baaebf/src/app/components/KcApp/KcApp.tsx#L84-L89) you have an example of how we changed the background to obtain:  

<p align="center">
    <i> <a href="https://datalab.sspcloud.fr">Customization using only CSS:</a> </i> 
    <br>
    <img src="https://github.com/InseeFrLab/keycloakify/releases/download/v0.3.8/keycloakify_after.gif">
</p>  

If my answer helped you, you can follow me on GitHub, I do appreciate it :)

**Comment by meimosor** [comment]
How to connect to the Docker version of Keycloak. Can you help？ @garronej

**Reply by garronej** [reply]
There's now a video tutorial about it https://www.youtube.com/watch?v=x3ux2JM1Bxk&embeds_referring_euri=https%3A%2F%2Fcdn.iframe.ly%2F&source_ve_path=MjM4NTE&feature=emb_title
